clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1310,6 +1310,70 @@ def GetGlobalOp : CIR_Op<"get_global",
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// GetMemberOp
+//===----------------------------------------------------------------------===//
+
+def GetMemberOp : CIR_Op<"get_member"> {
+  let summary = "Get the address of a member of a record";
+  let description = [{
+    The `cir.get_member` operation gets the address of a particular named
+    member from the input record.
+
+    It expects a pointer to the base record as well as the name of the member
+    and its field index.
```

Comment:
What does this do for members of inherited types?  is that encoded via the `record`?  Or are they 'flattened' and the index is consistent?

Does it seem sensible to have examples of inheritance? 

One thing that might be interesting if we don't encode it, is sometimes casts between types are legal if they have a 'common initial sequence' depending on language mode and use... so I wonder if encoding that (which, debatably, can change the common-initial-sequence rules) is worth-while some-day.  (This last part is lamenting/thinking, not necessariy immediately actionable).

Reply:
> Or are they 'flattened' and the index is consistent?

This one. For a bit of extra context: whenever something more complex is going on CIRGen takes care of it such that the final getmember is reliable.

> One thing that might be interesting if we don't encode it is sometimes casts between types are legal if they have a 'common initial sequence' depending on language mode and use

Indeed interesting. If this turns out to help CIRGen / optimization we should be all in for it in the future. Any compelling use case in mind?

Reply:
There is a bunch of 'legal' vs 'UB' things with common-initial sequence, particularly in C that might be interesting to opt based on, but I don't have any good ideas.

Reply:
gotcha, I usually try to keep a log of these ideas, I'll ad this one. If anything else come up let me know.